Cesium.CodeGen/Ir/BlockItems/ForStatement.cs

```diff
@@ -28,26 +30,35 @@ public ForStatement(Ast.ForStatement statement)
         IExpression? initExpression,
         IExpression testExpression,
         IExpression? updateExpression,
-        IBlockItem body)
+        IBlockItem body,
+        string breakLabel)
     {
         _initExpression = initExpression;
         _testExpression = testExpression;
         _updateExpression = updateExpression;
         _body = body;
+        _breakLabel = breakLabel;
     }
 
     public IBlockItem Lower(IDeclarationScope scope)
-        => new ForStatement(
-            _initExpression?.Lower(scope),
-            _testExpression.Lower(scope),
-            _updateExpression?.Lower(scope),
-            _body.Lower(scope));
+    {
+        var forScope = new ForScope((IEmitScope)scope);
+        var breakLabel = forScope.GetBreakLabel();
+        scope.AddLabel(breakLabel!);
```

Comment:
I think it's possible to declare `BreakLabel` as non-nullable in `ForScope`, to avoid `!` here.